### PR TITLE
Improve resource file support

### LINKED ISSUE #467: dotnet-compile should preserve resource assembly info in deps file
Right now, the deps file does not contain the resource assemblies. The data is in the lock file, we just need to transfer it to the deps file.

### PR DESCRIPTION
**Pull request by anurse**
* Add satellite assemblies to deps file with locale data
* Publish satellite assemblies to output during publish
* Copy satellite assemblies from project-to-project dependencies on
build and publish

Fixes #467 

/cc @pakrym @DamianEdwards 

Supersedes #2282

---
This change is [Reviewable](https://reviewable.io/reviews/dotnet/cli/2511)

### CODE REVIEW

**Comment by pakrym** [comment]
:shipit:

File: TestAssets/TestProjects/ResourcesTests/TestLibraryWithResources/project.json [hunk 1]
```diff
@@ -0,0 +1,11 @@
+{
+  "version": "1.0.0-*",
+  "dependencies": {
+    "NETStandard.Library": "1.5.0-rc2-24008",
```
Comment (brthor) [inline TestAssets/TestProjects/ResourcesTests/TestLibraryWithResources/project.json]: this should be 24015, upgrade went in last night.

Comment (brthor) [inline TestAssets/TestProjects/ResourcesTests/TestLibraryWithResources/project.json]: Or is there a reason for the downgrade?

Comment (anurse) [inline TestAssets/TestProjects/ResourcesTests/TestLibraryWithResources/project.json]: No reason, I'll fix it.

**Pull request merged**